src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -894,6 +894,7 @@
    (schema/required-key "trusted_facts") {(schema/required-key "values") {schema/Str schema/Any}}
    (schema/required-key "facts") {(schema/required-key "values") {schema/Str schema/Any}}
    (schema/required-key "variables") {(schema/required-key "values") (schema/either [{schema/Str schema/Any}] {schema/Str schema/Any})}
+   (schema/optional-key "target_variables") {(schema/required-key "values") [{schema/Str schema/Any}]}
```

Comment:
Made this optional for backwards compatibility. Also chose the `Array[Hash]` syntax since that seems to be the preferred way of specifying target var values.

Note that the Orch-side changes are basically to pass-in the `target_variables` to the request. I _think_ it consists of eliminating https://github.com/puppetlabs/orchestrator/blob/main/src/puppetlabs/orchestrator/bolt/plan_actions.clj#L335 and letting the compile endpoint handle the shadowing (and it will only be done on the `main` branch).

Reply:
I've tested that `target_variables` works as expected manually via `curl`.

Reply:
Where would I make the relevant doc updates for this?

Reply:
https://github.com/puppetlabs/osp-docs/blob/b0af18f7c554097a54e36f22d0403c592fe48272/server/puppet-api/v3/compile.md

Reply:
IMO the `plan_compile` specific options do not need to maintain backwards compatibility.